File: Prgrammers/가장_큰_수.java
import java.util.*;

public class 가장_큰_수 {
	public static void main(String[] args) {
		int [] numbers = {3, 30, 34, 5, 9};
		solution(numbers);
	}
	public static String solution(int[] numbers) {
        String answer = "";
        String [] arr = new String[numbers.length];
        
        for(int i=0; i<numbers.length; i++)
        	arr[i] = Integer.toString(numbers[i]);
        	        
        Arrays.sort(arr, new Comparator<String>() {
			@Override
			public int compare(String o1, String o2) {
				return (o2+o1).compareTo(o1+o2) ;
			}
        });
        
        StringBuilder sb = new StringBuilder();
		
        for(String s : arr)
        	sb.append(s);

        return sb.toString();
    }
}

File: Prgrammers/c++/가장_큰_수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool check(string &a, string &b){
  return a+b > b+a;
}

string solution(vector<int> numbers) {
    string answer = "";

    vector <string> arr;
    for(auto iter : numbers)
      arr.push_back(to_string(iter));
    
    sort(arr.begin(), arr.end(), check);

    for(auto it : arr)
      answer += it;
    
    if(answer.at(0)=='0') return "0";
    else return answer;
}